File: java/pe/src/test/java/ktn/pe/Pe3Test.java
package ktn.pe;

import static org.junit.Assert.*;
import static org.hamcrest.CoreMatchers.*;

import java.util.ArrayList;
import java.util.List;

import ktn.pe.util.PeUtils;

import org.junit.Test;

public class Pe3Test {

    @Test
    public void test() {
        List<Long> divisors = new ArrayList<>();

        PeUtils.divisors(divisors, 0L);
        assertTrue("0 []", divisors.isEmpty());
        divisors.clear();

        PeUtils.divisors(divisors, 12L);
        assertThat("12 [2,2,3]", divisors, hasItems(2L, 2L, 3L));
        divisors.clear();

        PeUtils.divisors(divisors, 97L);
        assertThat("97 [97]", divisors, hasItems(97L));
        divisors.clear();
    }

}

File: kt/pe/src/test/kotlin/ktn/pe/Pe3Test.kt
package ktn.pe

import org.junit.Assert.*
import org.hamcrest.CoreMatchers.*

import java.util.ArrayList

import org.junit.Test

class Pe3Test {

    @Test
    fun test() {
        val divisors = ArrayList<Long>()

        PeUtils.divisors(divisors, 0L)
        assertTrue("0 []", divisors.isEmpty())
        divisors.clear()

        PeUtils.divisors(divisors, 12L)
        assertThat<List<Long>>("12 [2,2,3]", divisors, hasItems(*arrayOf(2L, 2L, 3L)))
        divisors.clear()

        PeUtils.divisors(divisors, 97L)
        assertThat<List<Long>>("97 [97]", divisors, hasItems(*arrayOf(97L)))
        divisors.clear()
    }

}
